src/codegen/compile.ml

```diff
@@ -10941,31 +11382,99 @@ and compile_prim_invocation (env : E.t) ae p es at =
 
 
   | OtherPrim "popcnt8", [e] ->
-    SR.Vanilla,
-    compile_exp_vanilla env ae e ^^
+    SR.UnboxedWord32 Type.Nat8,
+    compile_exp_as env ae (SR.UnboxedWord32 Type.Nat8) e ^^
     G.i (Unary (Wasm.Values.I32 I32Op.Popcnt)) ^^
     TaggedSmallWord.msb_adjust Type.Nat8
   | OtherPrim "popcnt16", [e] ->
-    SR.Vanilla,
-    compile_exp_vanilla env ae e ^^
+    SR.UnboxedWord32 Type.Nat16,
+    compile_exp_as env ae (SR.UnboxedWord32 Type.Nat16) e ^^
     G.i (Unary (Wasm.Values.I32 I32Op.Popcnt)) ^^
     TaggedSmallWord.msb_adjust Type.Nat16
+  | OtherPrim "popcntInt8", [e] ->
+    SR.UnboxedWord32 Type.Int8,
+    compile_exp_as env ae (SR.UnboxedWord32 Type.Int8) e ^^
+    compile_shrU_const (TaggedSmallWord.shift_of_type Type.Int8) ^^
```

Comment:
I am not sure whether this shift is needed (kind of a `lsb_adjust Nat8` skipping the sign extension). It could also be used for `Nat8`. Maybe we could add a comment stating that the bit counting is independent of whether the payload resides in LSB or MSB. The only important part is not to extend the sign bit...

Reply:
Ditto for popcntInt8. Fixed both.